File: 2191-sort-the-jumbled-numbers/2191-sort-the-jumbled-numbers.cpp
class Solution {
public:
    vector<int> sortJumbled(vector<int>& mapping, vector<int>& nums) {
        int count, mappedNum, num;
        unordered_map<int, int> mappedNums;
        for(int i = 0; i < nums.size(); i++) {
            count = 0;
            mappedNum = 0;
            num = nums[i];
            while(num != 0) {
                mappedNum += mapping[num % 10] * pow(10, count++);
                num /= 10;
            }
            mappedNums.insert({nums[i], nums[i] != 0 ? mappedNum : mapping[0]});
        }
        
        sort(nums.begin(), nums.end(), [&](int a, int b) {
            return mappedNums[a] <  mappedNums[b];
        });
            
        return nums;
    }
};

File: 2191-sort-the-jumbled-numbers/2191-sort-the-jumbled-numbers.py
class Solution:
    def sortJumbled(self, mapping: List[int], nums: List[int]) -> List[int]:
        tmp, count = 0, 0
        for i, x in enumerate(nums):
            tmp, count = 0, 0
            if x == 0:
                tmp = mapping[0]
            else:
                while x != 0:
                    tmp += mapping[x % 10] * 10 ** count
                    x //= 10
                    count += 1
            nums[i] = [nums[i], tmp]
        
        return [x for x, _ in sorted(nums, key=lambda x: (x[1]))]